parser_library/src/workspaces/processor_file_impl.cpp

```diff
@@ -100,13 +102,14 @@ parse_result processor_file_impl::parse_macro(
     if (macro_cache_.load_from_cache(cache_key, ctx))
         return true;
 
+    const bool collect_hl = should_collect_hl(ctx.hlasm_ctx.get());
     auto a = std::make_unique<analyzer>(get_text(),
         analyzer_options {
             get_location(),
             &lib_provider,
             std::move(ctx),
             data,
-            get_lsp_editing() ? collect_highlighting_info::yes : collect_highlighting_info::no,
+            collect_hl ? collect_highlighting_info::yes : collect_highlighting_info::no,
```

Comment:
```suggestion
            should_collect_hl(ctx.hlasm_ctx.get())? collect_highlighting_info::yes : collect_highlighting_info::no,
```

To unify with other calls?